lib/classes/TradeOffer.js

```diff
@@ -799,18 +822,25 @@ TradeOffer.prototype.duplicate = function() {
 
 TradeOffer.prototype.setMessage = function(message) {
 	if (this.id) {
-		throw new Error("Cannot set message in an already-sent offer");
+		throw new Error('Cannot set message in an already-sent offer');
 	}
 
 	this.message = message.toString().substring(0, 128);
 };
 
 TradeOffer.prototype.setToken = function(token) {
 	if (this.id) {
-		throw new Error("Cannot set token in an already-sent offer");
+		throw new Error('Cannot set token in an already-sent offer');
 	}
 
 	this._token = token;
 };
 
+// A helper to use inside lib/polling.js
+TradeOffer.prototype.needsConfirmation = function() {
+	return this.state === ETradeOfferState.CreatedNeedsConfirmation
```

Comment:
What about these weird ifs? I'd suggest placing new line only before `OR`, you can easily see all conditions that way, in this case:
```JS
return this.state === ETradeOfferState.CreatedNeedsConfirmation
    || (this.state === ETradeOfferState.Active && this.confirmationMethod !== EConfirmationMethod.None);
```